hmp/models/fixedn.py

```diff
@@ -253,23 +253,28 @@ def fit(
                 )
             resetwarnings()
 
-        self.lkhs = np.array([x[0] for x in estimates])
+        lkhs = np.array([x[0] for x in estimates])
         if self.starting_points > 1 :
-            max_lkhs = np.argmax(self.lkhs)
+            max_lkhs = np.argmax(lkhs)
         else:
             max_lkhs = 0
-        self.lkhs = self.lkhs[max_lkhs]
-        self.magnitudes =  np.array(estimates[max_lkhs][1])
-        self.parameters = np.array(estimates[max_lkhs][2])
-        self.traces = np.array(estimates[max_lkhs][3])
-        self.param_dev = np.array(estimates[max_lkhs][4])
-
-        self.level_dict = level_dict
-        self.levels = levels
-        self.mags_map = mags_map
-        self.pars_map = pars_map
+            
+        if lkhs.sum() != -np.inf:
```

Comment:
I would reverse the if statement to save some indentation, so `if <error_cond>: raise Error()`.

Also, for consistency you should probably use `np.isneginf`?

Reply:
I implemented your suggestion and ready to merge!